docs/en/ingest-management/elastic-agent/grant-access-to-elasticsearch.asciidoc

```diff
@@ -0,0 +1,154 @@
+
+[discrete]
+[[create-api-key-standalone-agent]]
+== Create API keys for standalone agents
+
+NOTE: API keys are sent as plain-text, so they only provide security when used
+in combination with Transport Layer Security (TLS). Our
+{ess-product}[hosted {ess}] on {ecloud} provides secure, encrypted connections
+out of the box! For self-managed {es} clusters, refer to
+<<output-elasticsearch-pki-certs-authentication-settings>>.
+
+You can set API keys to expire at a certain time, and you can explicitly
+invalidate them. Any user with the `manage_api_key` or `manage_own_api_key`
+cluster privilege can create API keys.
+
+For security reasons, we recommend using a unique API key per {agent}. You
+can create as many API keys per user as necessary.
+
+To create an API key for {agent}:
+
+. In {kib}, navigate to *Stack Management > API keys* and click
+*Create API key*.
+
+. Enter a name for your API key and select *Restrict privileges*. In the role
+descriptors box, copy and paste the following JSON. This example creates an
+API key with privileges for ingesting logs, metrics, traces, and synthetics:
++
+[source,json]
+----
+{
+  "standalone_agent": {
+    "cluster": [
+      "monitor"
+    ],
+    "indices": [
+      {
+        "names": [
+          "logs-*-*", "metrics-*-*", "traces-*-*", "synthetics-*-*" <1>
+        ],
+        "privileges": [
+          "auto_configure", "create_doc"
+        ]
+      }
+    ]
+  }
+}
+----
+<1> Adjust this list to match the data you want to collect. For example, if
+you aren't using APM or synthetics, remove `"traces-*-*"` and `"synthetics-*-*"`
+from this list.
+
+. To set an expiration date for the API key, select *Expire after time* and input
+the lifetime of the API key in days.
+
+. Click *Create API key*.
++
+You'll see a message indicating that the key was created, along with the
+encoded key. By default, the API key is Base64 encoded, but that won't work for
+{agent}.
+
+. Click the down arrow next to Base64 and select *Beats*.
++
+[role="screenshot"]
+image::images/copy-api-key.png[Message with field for copying API key]
+
+. Copy the API key. You will need this for the next step, and you will not be
+able to view it again.
+
+. To use the API key, specify the `api_key` setting in the `elastic-agent.yml`
+file. For example:
++
+[source,yaml]
+----
+[...]
+outputs:
+  default:
+    type: elasticsearch
+    hosts:
+      - 'https://da4e3a6298c14a6683e6064ebfve9ace.us-central1.gcp.cloud.es.io:443'
+    api_key: _Nj4oH0aWZVGqM7MGop8:349p_U1ERHyIc4Nm8_AYkw <1>
+[...]
+----
+<1> The format of this key is `<id>.<key>`. Base64 encoded API keys are not
+currently supported in this configuration.
+ 
+For more information about creating API keys in {kib}, see
+{kibana-ref}/api-keys.html[API Keys].
+
+[discrete]
+[[create-role-standalone-agent]]
+== Create a standalone agent role
+
+Although it's recommended that you use an API key instead of a username and
```

Comment:
Note: This does NOT block this PR

@nimarezainia I wonder if we should even document this here as we don't recommend this approach.

Reply:
I think we should at least for now because the standalone policy that gets generated shows those settings.